weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallsPage/PivotRunsTable.tsx

```diff
@@ -0,0 +1,335 @@
+import _ from 'lodash';
+import React, {useEffect, useMemo, useState} from 'react';
+
+import {parseRef} from '../../../../../../react';
+import {flattenObject} from '../../../Browse2/browse2Util';
+import {SpanWithFeedback} from '../../../Browse2/callTree';
+import {
+  buildTree,
+  DataGridColumnGroupingModel,
+} from '../../../Browse2/RunsTable';
+import {SmallRef} from '../../../Browse2/SmallRef';
+import {StyledDataGrid} from '../../StyledDataGrid';
+import {renderCell} from '../util';
+
+export type WFHighLevelPivotSpec = {
+  rowDim: string | null;
+  colDim: string | null;
+};
+
+export const PivotRunsView: React.FC<{
+  loading: boolean;
+  runs: SpanWithFeedback[];
+  pivotSpec: Partial<WFHighLevelPivotSpec>;
+  onPivotSpecChange: (spec: Partial<WFHighLevelPivotSpec>) => void;
+}> = props => {
+  const pivotRowDim = props.pivotSpec.rowDim;
+  const pivotColDim = props.pivotSpec.colDim;
+
+  const [pivotRowOptions, setPivotRowOptions] = useState<string[]>([]);
+  const [pivotColOptions, setPivotColOptions] = useState<string[]>([]);
+
+  useEffect(() => {
+    const runs = props.runs;
+    if (runs.length === 0) {
+      return;
+    }
+    const firstRun = runs[0];
+    const options: string[] = [];
+    Object.entries(firstRun.inputs).forEach(([key, value]) => {
+      if (
+        typeof value === 'string' &&
+        value.startsWith('wandb-artifact:///') &&
+        !key.startsWith('_')
+      ) {
+        options.push('inputs.' + key);
+      }
+    });
+    setPivotRowOptions(options);
+    setPivotColOptions(options);
+    if (
+      props.pivotSpec.colDim === undefined &&
+      props.pivotSpec.rowDim === undefined &&
+      options.length > 1
+    ) {
+      if (options[0] === 'inputs.self') {
+        props.onPivotSpecChange({
+          rowDim: options[1],
+          colDim: options[0],
+        });
+      } else {
+        props.onPivotSpecChange({
+          rowDim: options[0],
+          colDim: options[1],
+        });
+      }
+    }
+  }, [props]);
+
+  return (
+    <Box
+      sx={{
+        height: '100%',
+        overflow: 'hidden',
+        display: 'flex',
+        flexDirection: 'column',
+      }}>
+      <Box
+        sx={{
+          flex: '0 0 auto',
+          width: '100%',
+          transition: 'width 0.1s ease-in-out',
+          display: 'flex',
+          flexDirection: 'row',
+          overflowX: 'auto',
+          overflowY: 'hidden',
+          alignItems: 'center',
+          gap: '8px',
+          p: 1,
+          '& li': {
+            padding: 0,
+            minWidth: '150px',
+          },
+          '& input, & label, & .MuiTypography-root': {
+            fontSize: '0.875rem',
+          },
+        }}>
+        <Typography
+          style={{width: '38px', textAlign: 'center', flex: '0 0 auto'}}>
+          Pivot
+        </Typography>
+        <ListItem>
+          <FormControl fullWidth>
+            <Autocomplete
+              size={'small'}
```

Comment:
I like to save characters and just use size="small"

Reply:
👍